src/analysis/inner_product_space/of_norm.lean

```diff
@@ -0,0 +1,346 @@
+/-
+Copyright (c) 2020 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+
+import topology.algebra.algebra
+import analysis.inner_product_space.basic
+
+/-!
+# Inner product space derived from a norm
+
+This file defines an `inner_product_space` instance from a norm that respects the
+parallellogram identity. The parallelogram identity is a way to express the inner product of `x` and
+`y` in terms of the norms of `x`, `y`, `x + y`, `x - y`.
+
+## Main results
+
+- `inner_product_space.of_norm`: a normed space whose norm respects the parallellogram identity,
+  can be seen as an inner product space.
+
+## Tags
+
+inner product space, Hilbert space, norm
+
+## References
+
+- http://www.mathematik.uni-muenchen.de/~michel/jordan-von_neumann_-_parallelogram_identity.pdf
+- https://math.stackexchange.com/questions/21792/norms-induced-by-inner-products-and-the-parallelogram-law
+- https://math.dartmouth.edu/archive/m113w10/public_html/jordan-vneumann-thm.pdf
```

Comment:
Do any of these belong in `references.bib`?

Reply:
One of them, I think